src/services/notifier/utils/updaterUtils.ts

```diff
@@ -50,25 +54,85 @@ export async function deactivateDeletedPlansForProvider (provider: string, url:
+  const {
+    hash,
+    currency: {
+      address: {
+        value: tokenAddress
+      }
+    },
+    price,
+    expirationDate,
+    id: subscriptionId,
+    paid,
+    status,
+    notificationBalance,
+    subscriptionPlanId,
+    previousSubscription: previousSubscriptionModel,
+    topics,
+    signature,
+    userAddress: consumer
+  } = subscriptionDTO
+  const tokenSymbol = getTokenSymbol(tokenAddress, SupportedServices.NOTIFIER).toLowerCase()
+  const previousSubscriptionHash = previousSubscriptionModel?.hash
+
+  const subscription = {
+    providerId: provider,
+    price: new BigNumber(price),
+    rateId: tokenSymbol,
+    expirationDate: new Date(expirationDate),
+    hash,
+    consumer,
+    subscriptionId,
+    paid,
+    status,
+    notificationBalance,
+    subscriptionPlanId,
+    previousSubscription: previousSubscriptionHash,
+    topics,
+    signature
+  }
+
+  return subscription
+}
+
+const findOrCreateSubscription = async (subscriptionDTO: SubscriptionDTO, url: string) => {
+  const { hash, status, paid, notificationBalance, expirationDate } = subscriptionDTO
+  const found = await SubscriptionModel.findOne({ where: { hash } })
+
+  if (found) {
+    return SubscriptionModel.update(
+      { status, paid, notificationBalance, expirationDate },
+      { where: { hash } })
+  }
+  const providerModel = await ProviderModel.findOne({ where: { url } })
+  const provider = providerModel?.provider
+
+  if (!provider) {
+    throw new Error(`Provider model with url ${url} not found.`)
+  }
+
+  const subscription = buildSubscriptionFromDTO(subscriptionDTO, provider)
+  return SubscriptionModel.create(subscription)
+}
+
 export const updateSubscriptionsBy = async (
-  providerUrl: string, consumerAddress: string, subsHashesToUpdate: string[]
-): Promise<void> => {
+  providerUrl: string, consumerAddress: string): Promise<void> => {
   const [host, port] = providerUrl.split(/(?::)(\d*)$/, 2)
   const svcProvider = new NotifierSvcProvider({ host, port })
   try {
-    const subscriptionsDTO: any[] = await svcProvider.getSubscriptions(consumerAddress, subsHashesToUpdate)
-    await Promise.all(subscriptionsDTO.map(({
-      hash,
-      status,
-      paid,
-      notificationBalance,
-      expirationDate
-    }) =>
-      SubscriptionModel.update(
-        { status, paid, notificationBalance, expirationDate },
-        { where: { hash } }
-      ).catch(error => logger.warn(`Unable to update subscription with hash ${hash} in the database`, error))
-    ))
+    const subscriptionsDTO: Array<SubscriptionDTO> = await svcProvider.getSubscriptions(consumerAddress)
+
+    await Promise.all(
+      subscriptionsDTO.map(
+        (subscription) => findOrCreateSubscription(
+          subscription, providerUrl
+        ).catch(
+          error => logger.warn(`Unable to update or create subscription with hash ${subscription.hash} in the database`, error)
+        )
```

Comment:
Why would a simple `SubscriptionModel.findOrCreate` not suffice?
something like:
```suggestion
        (subscription) => SubscriptionModel
          .findOrCreate(
            buildSubscriptionFromDTO(subscriptionDTO, provider),
            { where: { hash: subscription.hash }}
          ).catch(
            error => logger.warn(`Unable to update or create subscription with hash ${subscription.hash} in the database`, error)
          )
        )
```

Reply:
Because when finding it, we need to take action and also update `{ status, paid, notificationBalance, expirationDate }` fields